crates/bins/src/bin/datadog-static-analyzer.rs

```diff
@@ -610,6 +613,25 @@ fn main() -> Result<()> {
                         }
                         JS_RUNTIME.replace(opt);
 
+                        // Test file detection is only output for SARIF.
+                        if !results.is_empty()
+                            && matches!(configuration.output_format, OutputFormat::Sarif)
+                            && !path_metadata.contains_key(relative_path.as_ref())
+                        {
+                            let cloned_path_str = relative_path.to_string();
+                            let metadata = if is_test_file(
+                                *language,
+                                file_content.as_ref(),
+                                std::path::Path::new(&cloned_path_str),
+                                None,
+                            ) {
+                                Some(ArtifactClassification { is_test_file: true })
+                            } else {
+                                None
+                            };
+                            path_metadata.insert(cloned_path_str, metadata);
+                        }
+
```

Comment:
1. Do not think it should be sarif-specific.
2. Can we try to extract this into a function that can be reused by the git hooks?